src/DataProtection/DataProtection/test/Microsoft.AspNetCore.DataProtection.Tests/XmlEncryption/XmlEncryptionExtensionsTests.cs

```diff
@@ -49,6 +49,37 @@ public void DecryptElement_RootNodeRequiresDecryption_Success()
         XmlAssert.Equal("<newNode />", retVal);
     }
 
+    [Fact]
+    public void DecryptElement_MissingDecryptorType_Success()
+    {
+        // Arrange
+        // We want to simulate an error loading the specified decryptor type, i.e.
+        // Could not load file or assembly 'Azure.Extensions.AspNetCore.DataProtection.Keys,
+        // Version=1.2.2.0, Culture=neutral, PublicKeyToken=92742159e12e44c8' or one of its dependencies.
+        XmlEncryptionExtensions._getType = _ => throw new TypeLoadException();
```

Comment:
I worry about static state like this impacting other unit tests. It's not being reset so unit tests run afterwards would run this func. And if it was reset, there is still the possibility of concurrent tests.

Is it necessary? Can a value be specified that would trigger TypeLoadException without having to override how the GetType call worked.

Reply:
I did try to find a way to produce a `FileLoadException` or `TypeLoadException` by modifying the type, namespace, assembly, version etc. of the `decryptorType`, but wasn't able to. We're probably seeing this more because we're using .NET Framework with (semi-)manual binding redirects, while .NET (Core) seems much more forgiving.

If you have any suggestions, I'm all ears 😅

Reply:
.NET Core probably changed the behavior so that `throwOnError` was always respected.

Not having a unit test is one solution. However, when I wrote this code I didn't consider that an error could be thrown here. Catching it is a band aid.

Maybe the solution here is to match to these known types from a string without using `Type.GetType`. If the type name and assembly match, then we go directly to the known types. We'd be getting into the type name parsing business though, which is scary.

Reply:
> .NET Core probably changed the behavior so that `throwOnError` was always respected.

It's still documented to throw under the same circumstances, but .NET (Core) is much more resilient when it comes to resolving multiple versions of the same assemblies.

> Maybe the solution here is to match to these known types from a string without using Type.GetType. If the type name and assembly match, then we go directly to the known types. We'd be getting into the type name parsing business though, which is scary.

Yeah, why were the type checks introduced in the first place? I see there's a similar issue in `XmlKeyManager.CreateDeserializer`.

I have a crazy idea; what about an internal (or maybe even public?) interface

```csharp
internal interface ITypeForwardingActivator : IActivator
{
    bool TryForwardType(string decryptorTypeName, [NotNullWhen(true)] out Type? type);
}
```

which could be checked by pattern matching

```csharp
private static IXmlDecryptor CreateDecryptor(IActivator activator, string decryptorTypeName)
{
    if (activator is ITypeForwardingActivator typeForwarder && typeForwarder.TryForwardType(decryptorTypeName, out var type))
    {
        if (type == typeof(DpapiNGXmlDecryptor))
        {
            return activator.CreateInstance<DpapiNGXmlDecryptor>(decryptorTypeName);
        }
        else if (type == typeof(DpapiXmlDecryptor))
        {
            return activator.CreateInstance<DpapiXmlDecryptor>(decryptorTypeName);
        }
        else if (type == typeof(EncryptedXmlDecryptor))
        {
            return activator.CreateInstance<EncryptedXmlDecryptor>(decryptorTypeName);
        }
        else if (type == typeof(NullXmlDecryptor))
        {
            return activator.CreateInstance<NullXmlDecryptor>(decryptorTypeName);
        }
    }

    return activator.CreateInstance<IXmlDecryptor>(decryptorTypeName);
}
```

This would be implemented by the default `TypeForwardingActivator` and reuse the same existing logic.

Then the unit test could provide its own faked/mocked activator that returns `false` from `TryForwardType` throug the `IActivator` argument?

Reply:
I've pushed a commit to show what it would look like 😄

Reply:
But the "what are we even testing at this point?" question is becoming more and more relevant 😆